test/sql/alter/alter_col/test_add_primary_key.test

```diff
@@ -0,0 +1,206 @@
+# name: test/sql/alter/alter_col/test_add_primary_key.test
+# description: Test ALTER TABLE table ADD PRIMARY KEY (column)
+# group: [alter_col]
+
+statement ok
+PRAGMA enable_verification
+
+# Scenario #1: single column pk
+statement ok
+CREATE TABLE test1 (i INTEGER, j INTEGER)
+
+statement ok
+INSERT INTO test1 VALUES (1, 1)
+
+statement ok
+ALTER TABLE test1 ADD PRIMARY KEY (j)
+
+# Check data is preserved
+query II
+SELECT * FROM test1
+----
+1	1
+
+statement ok
+INSERT INTO test1 VALUES (1, 2)
+
+statement error
+INSERT INTO test1 VALUES (2, 1)
+----
+Duplicate key "j: 1" violates primary key constraint
+
+statement error
+INSERT INTO test1 VALUES (2, NULL)
+----
+NOT NULL constraint failed: test1.j
+
+# altering a second time should fail, only one pk is allowed per table
+statement error
+ALTER TABLE test1 ADD PRIMARY KEY (i)
+----
+Catalog Error: table "test1" can have only one primary key, and already has PRIMARY KEY(j).
+
+# even if it's an exact duplicate
+statement error
+ALTER TABLE test1 ADD PRIMARY KEY (j)
+----
+Catalog Error: table "test1" can have only one primary key, and already has PRIMARY KEY(j).
+
+# Scenario #2: multiple column pk
+statement ok
+CREATE TABLE test2 (i INTEGER, j INTEGER, d TEXT)
+
+statement ok
+ALTER TABLE test2 ADD PRIMARY KEY (i, j)
+
+statement ok
+INSERT INTO test2 VALUES (1, 1, 'foo')
+
+statement ok
+INSERT INTO test2 VALUES (1, 2, 'bar')
+
+statement error
+INSERT INTO test2 VALUES (1, 2, 'oops')
+----
+Duplicate key "i: 1, j: 2" violates primary key constraint
+
+statement error
+INSERT INTO test2 VALUES (NULL, 2, 'nada')
+----
+NOT NULL constraint failed: test2.i
+
+# Scenario #3: primary key already exists
+statement ok
+CREATE TABLE test3 (i INTEGER PRIMARY KEY, j INTEGER)
+
+statement error
+ALTER TABLE test3 ADD PRIMARY KEY (i, j)
+----
+Catalog Error: table "test3" can have only one primary key, and already has PRIMARY KEY(i).
+
+# Scenario #4: column does not exist
+statement ok
+CREATE TABLE test4 (i INTEGER, j INTEGER)
+
+statement error
+ALTER TABLE test4 ADD PRIMARY KEY (i_do_not_exist)
+----
+Table "test4" does not have a column named "i_do_not_exist"
+
+# Scenario #5: table does not exist
+statement error
+ALTER TABLE i_do_not_exist ADD PRIMARY KEY (i, j)
+----
+Table with name i_do_not_exist does not exist!
+
+# Scenario #6: add pk to table containing data invalidating the constraint (this should fail)
+statement ok
+CREATE TABLE test6 (i INTEGER, j INTEGER)
+
+statement ok
+INSERT INTO test6 VALUES (1, 10), (2, 20), (3, 30), (1, 100), (NULL, 0)
+
+statement error
+ALTER TABLE test6 ADD PRIMARY KEY (i)
+----
+PRIMARY KEY or UNIQUE constraint violated: duplicate key "1"
```

Comment:
Let's divide this into two separate scenarios:
- Creation failure due to duplicate data.
- Creation failure due to `NULL`.

Reply:
Test added.